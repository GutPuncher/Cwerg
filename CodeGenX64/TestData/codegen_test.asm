
.mem COUNTER 4 RW
.data 4 [0]

.fun TestMoveImmediates NORMAL [] = [A64]
  .reg U32 [$r0 $r1 $r2 $r3 bitu]
  .reg S32 [$r0_S32 bits]
  .reg S64 [$r10 $r11 $12]
  .reg F32 [$f0 $f1 $f2]
  .reg F64 [$f10 $f11 $f12]
  .reg U8 [$r0_U8 %U8_857]
  .reg A64 [addr1 addr2]
.bbl int
    add $r10@rax $r10@rax 0x12345678:S64
    and $r1@rdx $r1@rdx 42
    xor $r2@rdx $r2@rdx $r3@rcx
    ld  $r2@rdx addr1@rdx 0x20:S32
    ld  $r2@rdx addr1@rdx $r0_U8@rdx
    lea addr1@rdx addr1@rdx $r0_U8@rdx
    conv bits@STK %U8_857@rcx
    conv bitu@STK %U8_857@rcx

.bbl flt
    add $f1@xmm0 $f1@xmm0 $f2@xmm1

    ret




