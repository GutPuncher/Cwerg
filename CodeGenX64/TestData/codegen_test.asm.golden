
INS: add $r10@rax $r10@rax 305419896  [S64 S64 S64]
PAT: reg:[S64 S64 S64]  op:[REG REG SIMM32]
    add_64_mr_imm32 rax 0x12345678

INS: and $r1@rdx $r1@rdx 42  [U32 U32 U32]
PAT: reg:[U32 U32 U32]  op:[REG REG UIMM32]
    and_32_mr_imm32 edx 0x2a

INS: xor $r2@rdx $r2@rdx $r3@rcx  [U32 U32 U32]
PAT: reg:[U32 U32 U32]  op:[REG REG REG]
    xor_32_r_mr edx ecx

INS: ld $r2@rdx addr1@rdx 32  [U32 A64 S32]
PAT: reg:[U32 A64 S32]  op:[REG REG SIMM32]
    mov_32_r_mbis32 edx rdx noindex 0 32

INS: ld $r2@rdx addr1@rdx $r0_U8@rdx  [U32 A64 U8]
PAT: reg:[U32 A64 U8]  op:[REG REG REG]
    movzx_64_8_r_mr rdx dl
    mov_32_r_mbis8 edx rdx rdx 0 0

INS: lea addr1@rdx addr1@rdx $r0_U8@rdx  [A64 A64 U8]
PAT: reg:[A64 A64 U8]  op:[REG REG REG]
    movzx_64_8_r_mr rdx dl
    lea_64_r_mbis8 rdx rdx rdx 0 0

INS: conv bits@STK %U8_857@rcx  [S32 U8]
PAT: reg:[S32 U8]  op:[SP_REG REG]
    movzx_32_8_r_mr ecx cl
    mov_32_mbis32_r rsp noindex 0 4 ecx

INS: conv bitu@STK %U8_857@rcx  [U32 U8]
PAT: reg:[U32 U8]  op:[SP_REG REG]
    movzx_32_8_r_mr ecx cl
    mov_32_mbis32_r rsp noindex 0 0 ecx

INS: add $f1@xmm0 $f1@xmm0 $f2@xmm1  [F32 F32 F32]
PAT: reg:[F32 F32 F32]  op:[REG REG REG]
    addss_x_mx xmm0 xmm1

INS: ret  []
    SPECIAL
